### PR TITLE
feat: added fs cache dir support as source

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✔                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✔                                                                     |

### CODE REVIEW

File: src/common/enums/sourceTypes.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export enum SourceTypes {
   GPKG = 'geopackage',
   S3 = 's3',
+  FS = 'file',
```
Comment (CL-SHLOMIKONCHA) [inline src/common/enums/sourceTypes.ts:4]: its 'file' - because mapproxy defined this value as file system "cache dir" inside the yaml file

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: src/layers/models/layersManager.ts [hunk 2]
```diff
@@ -171,18 +173,19 @@ class LayersManager {
...
-    if (filePathExtension === this.mapproxyConfig.cache.gpkgExt) {
+    if (cacheSource === SourceTypes.GPKG) {
       sourceProvider = new GpkgSource();
-    } else if (filePathExtension === '') {
+    } else if (cacheSource === SourceTypes.S3) {
       sourceProvider = new S3Source(container);
+    } else if (cacheSource === SourceTypes.FS) {
+      sourceProvider = new FSSource(container);
+    } else {
+      throw new Error('Invalid source provider due to invalid source path, available values: "geopackage", "s3", "file"');
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply src/layers/models/layersManager.ts:185]: added in 
f0ab226

File: openapi3.yaml [hunk 3]
```diff
@@ -238,10 +238,14 @@ components:
     layer:
       type: object
       required:
+        - cacheType
         - name
         - tilesPath
         - maxZoomLevel
       properties:
+        cacheType:
+          type: string
```
[the remark this answers is not in the document]
Comment (CL-SHLOMIKONCHA) [reply openapi3.yaml:247]: fixed in 
c31e262

File: src/layers/models/layersManager.ts [hunk 4]
```diff
@@ -171,18 +172,19 @@ class LayersManager {
...
   }
 
-  public getCacheType(sourcePath: string): IS3Source | IGpkgSource {
-    let sourceProvider: ICacheProvider | undefined;
-    const filePathExtension = getFileExtension(sourcePath);
+  public getCacheType(cacheSource: string, sourcePath: string): ICacheSource {
+    let sourceProvider: ICacheProvider;
 
-    if (filePathExtension === this.mapproxyConfig.cache.gpkgExt) {
+    if (cacheSource === SourceTypes.GPKG) {
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts:178]: why not to use switch?

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: src/layers/models/layersManager.ts (hunk 4 above)
Comment (CL-SHLOMIKONCHA) [reply src/layers/models/layersManager.ts]: fixed in 9f1cc68

**Review by asafMasa — APPROVED** [review]

**Pull request merged**